### PR TITLE
fix(fonts): update fonts dependencies

### PR DESCRIPTION
**Pull request by ylemkimon**
- Update base image to Ubuntu 20.04
- Use precompiled `ttfautohint` (1.8.3)
  - Remove build-related tools: `wget`, `man-db`, `pkg-config`
- Use Python 3
- Update `fonttools` to 4.21.1

There are subpixel changes to the screenshots: [diff.zip](https://github.com/KaTeX/KaTeX/files/6229538/diff.zip).

### CODE REVIEW

**Comment by ylemkimon** [comment]
This PR is ready for review.

**Comment by edemaine** [comment]
@ylemkimon I'm happy to take a look at this now. Could you resolve the conflict / update the screenshots?

**Comment by edemaine** [comment]
@ylemkimon It would be nice to get font building working again. Do you have some time to look at this, or should I try to fix it myself?

**Comment by ylemkimon** [comment]
@edemaine Sorry for the delay. This PR is ready for review.

File: src/fontMetricsData.js [hunk 1]
```diff
@@ -1133,17 +1133,17 @@ export default {
         "8901": [-0.05555, 0.44445, 0, 0, 0.27778],
         "8902": [-0.03472, 0.46528, 0, 0, 0.5],
         "8904": [0.005, 0.505, 0, 0, 0.9],
-        "8942": [0.03, 0.9, 0, 0, 0.278],
-        "8943": [-0.19, 0.31, 0, 0, 1.172],
-        "8945": [-0.1, 0.82, 0, 0, 1.282],
+        "8942": [0.03, 0.903, 0, 0, 0.278],
+        "8943": [-0.19, 0.313, 0, 0, 1.172],
+        "8945": [-0.1, 0.823, 0, 0, 1.282],
```
Comment (edemaine) [inline src/fontMetricsData.js:1138]: These metric changes are presumably why there are subpixel differences in the screenshots.  They seem to be becoming more accurate, though, so presumably an improvement.

**Review by edemaine — APPROVED** [review]
I finally reviewed this, testing on a Linux machine.  Everything looks great, in particular generating the exact same font files, except for two commits which I added already.  Could you review them and then merge?  Here is what I changed:

* The code did `cd "$(dirname "$0")/../.."` and then referred to the path `$(dirname $0)` again.  This doesn't work when the path is relative (e.g. `./buildFonts.sh` or even `dockers/font/buildFonts.sh`). I replaced the latter occurrences with `dockers/fonts`.
* There was an uncalled function that claimed to deal with arguments, but no arguments seem to actually be supported.  (We could support `-h`, but why bother if no arguments are supported. 🙂 )

In either case, feel free to replace with more suitable code.  Thanks for these improvements!

**Comment by edemaine** [comment]
It seems that `buildFonts.sh` doesn't run `buildMetrics.sh`.  Do we not have a Dockerized `buildMetrics.sh`?  It would be nice to include, perhaps in a future PR.

**Comment by edemaine** [comment]
Another update: Should we add scripts for `build:fonts` and `build:metrics` or something along those lines?  Having them in the `scripts` list makes it more obvious that they exist.

**Comment by ylemkimon** [comment]
@edemaine Thank you very much!

**Comment by edemaine** [comment]
Oops, maybe this should have been `chore` instead of `fix`, but not a big deal. 🙂 We'll have a bunch more releases with actual font-based features soon.

**Comment by ylemkimon** [comment]
@edemaine Since fonts are updated, a patch release makes sense.

**Pull request merged**